### PR TITLE
initial feature set

### PR DESCRIPTION
**Pull request by blizzz**
Adds two buttons in the "Personal Info" section of Nextcloud. At the bottom. They look a bit lost, but I guess it is the fitting place.

screenshot_20180517_224228

A click on the button disables it and brings a spinner:

screenshot_20180517_224332

On success it stays disabled, the text updated:

screenshot_20180517_224407

And then we also have error reporting feedback if somethings wrong:

screenshot_20180517_224506

In the backend, any admin (so member of admin group) who has set an email, is getting notified separately. If no email could have been sent successfully, it is also shown as error.

The endpoints for requesting export or deletion are OCS API (authenticated), so it can be done programmatically by the user. There is not rate limiting though… if you hate your admin(s) you could spam them).

This meets basic requirements.

Potential to make it more awesome:
* activities ("You requested an export on Tue")
* notifications for the admin(s) ("User X requests deletion")
* tracking of the state of the requests
  * with an admin panel
  * this way you can limit requesting/sending limit to just one active request

### CODE REVIEW

**Comment by blizzz** [comment]
@jospoortvliet @jancborchardt

**Comment by jancborchardt** [comment]
Let’s simplify the wording:
- Account
- Export data
- Delete account

"Delete account" should get a `border-color: $color-error`

Or why is it "Request"? Data export should work right away, no? And when deleting the account, we could immediately deactivate it and say that it could take up to X days to remove it (and when logging in during that time, it will be reactivated), similar to how other services do.

**Comment by MorrisJobke** [comment]
> Or why is it "Request"? Data export should work right away, no? And when deleting the account, we could immediately deactivate it and say that it could take up to X days to remove it (and when logging in during that time, it will be reactivated), similar to how other services do.

This is not automatic. This is only to trigger an email to the admins. It's pure manual work, because it also affects backups, access logs etc, which we can't control.

**Comment by blizzz** [comment]
> This is not automatic. This is only to trigger an email to the admins. It's pure manual work, because it also affects backups, access logs etc, which we can't control.

perhaps this is not getting clear from the UI. Perhaps it needs some better explanation, either a more clear button text or some short paragraph below?

@jancborchardt changed the heading to "Account" anyway and also adapted border color to the deletion request.

**Comment by jospoortvliet** [comment]
Perhaps add this sentence under 'account':
You can request an export of your data or account deletion from the system administrator. This can take up to 30 days to be executed.

**Comment by blizzz** [comment]
@jospoortvliet thx, added!

screenshot_20180518_172040

**Comment by jospoortvliet** [comment]
:+1:

**Comment by MariusBluem** [comment]
- Who sets those 30 days as maximal period?
- Wouldn’t this fit better in the Security settings (or maybe then rename it to „Security & Privacy“)?

**Comment by jospoortvliet** [comment]
The 30 days is required by the GDPR. I'd not put it in security but close to the actual personal settings.

**Comment by rullzer** [comment]
@jancborchardt 

>Or why is it "Request"? Data export should work right away, no?

As @MorrisJobke it is also in backups/logs whatever.
But also it would timeout horribly if you all of a sudden tried to create a big zipfile containing all your data (possibly several GBs).

**Comment by blizzz** [comment]
> As @MorrisJobke it is also in backups/logs whatever.

Also, we have no idea what arbitrary apps are storing wherever.

>  But also it would timeout horribly if you all of a sudden tried to create a big zipfile containing all your data (possibly several GBs).

In the long term we could have some background jobs that gathers the known stuff and prepares an archive… if that becomes a requirement. Still this leaves the admin to collect everything that is stored elsewhere.

File: templates/Settings/personal.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+style('data_request', 'style');
+script('data_request', ['init', 'app']);
+
+
+?>
+
+<div id="data-request" class="section">
+	<h2><?php p($l->t('Account')); ?></h2>
+	<p><?php p($l->t('You can request an export of your data or account deletion from the system administrator. This can take up to 30 days to be executed.')); ?></p>
```
Comment (jancborchardt) [inline templates/Settings/personal.php]: Cut the "to be executed", too technical language.

Also, add the class "settings-hint" to the p, just like it is in the other explanatory sentences in settings:
https://github.com/nextcloud/server/blob/master/settings/templates/settings/personal/security.php#L36

**Comment by blizzz** [comment]
Oh, and for completion, the emails (in case you didn't read the text in the source): 

screenshot_20180522_144729

screenshot_20180522_144806

**Comment by blizzz** [comment]
any further complains go as issues por favor ¯\_(ツ)_/¯
